pkg/controller/servicebinding/secret.go

```diff
@@ -27,17 +31,47 @@ func (s *secret) buildResourceClient() dynamic.ResourceInterface {
 	return s.client.Resource(gvr).Namespace(s.ns)
 }
 
-// createOrUpdate will take informed payload and either create a new secret or update an existing
-// one. It can return error when Kubernetes client does.
-func (s *secret) createOrUpdate(payload map[string][]byte, ownerReference metav1.OwnerReference) (*unstructured.Unstructured, error) {
+func secretDataHash(data map[string]byte) string {
+	keys := make([]string, 0, len(payload))
+	for k := range payload {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	hash := sha1.New()
+	for _, k := range keys {
+		hash.Write([]byte(k))
+		hash.Write(data[k])
+	}
+	return hex.EncodeToString(hash.Sum(nil))
+}
+
+func uniqueSecret(name string, payload map[string][]byte) (string, error) {
```

Comment:
let's make the intention here a bit clearer, the function returns the secret name, so why do not we then declare it on `secret` struct?

```
func (s *secret) Name() string {
if s.fullName == nil {
  s.fullName = s.fullName = s.name + "-" + secretDataHash(s.data)
}
return s.fullName
}
```

Reply:
This is addressed.